Lib/unittest/test/test_case.py

```diff
@@ -1950,5 +1953,234 @@ def test2(self):
+            ),
+            Case(
+                line_length_factor=2,
+                num_lines_factor=1,
+                extent_differing=1,
+                expect_unified_diff_at = 28,
+            ),
+
+            # scale width and length by ratios of 3:1
+            Case(
+                line_length_factor=1,
+                num_lines_factor=3,
+                extent_differing=1,
+                expect_unified_diff_at = 16,
+            ),
+            Case(
+                line_length_factor=3,
+                num_lines_factor=1,
+                extent_differing=1,
+                expect_unified_diff_at = 24,
+            ),
+
+            # # scale by ratios of 3:1, with only 40% differing
+            Case(
+                line_length_factor=3,
+                num_lines_factor=1,
+                extent_differing=1,
+                expect_unified_diff_at = 24,
+            ),
+            Case(
+                line_length_factor=1,
+                num_lines_factor=3,
+                extent_differing=0.4,
+                expect_unified_diff_at = 16,
+            ),
+            Case(
+                line_length_factor=3,
+                num_lines_factor=1,
+                extent_differing=0.4,
+                expect_unified_diff_at = 23,
+            ),
+        )
+
+        # --- Execute test cases
+        # ----------------------
+
+        def run_case(case: Case, N):
+            """Given one of the test cases above, execute the test case for a
+            given `N` constant value. Check if the test has passed as
+            specified."""
+
+            # we are working out way towards _heuristic_diff(foo, bar)
+
+                    # --- Construct Differing Strings ---
+
+            # construct foo. Double line count because bar will have twice
+            # as many lines (lines of 'a' and lines of 'b')
+            foo = (
+                ('a' * N * (case.line_length_factor),)  # create line
+                * (N * case.num_lines_factor * 2)       # duplicate line
+            )
+
+            # construct bar
+            bar_a_line = ('a' * (N * case.line_length_factor))
+            bar_b_line = ('b' * (N * case.line_length_factor))
+            bar_a_lines = ((bar_a_line,) * (N * case.num_lines_factor))
+            if case.extent_differing != 1:
+                # diminish the amount of 'b' by case.extent_differing, and add
+                # additional 'a' at the end as padding
+                bar_b_lines = ((bar_b_line,)
+                               * int(N
+                                     * case.num_lines_factor
+                                     * case.extent_differing))
+                bar_a_padding = ((bar_a_line,)
+                                 * int(N
+                                       * case.num_lines_factor
+                                       * (1 - case.extent_differing)))
+                bar = tuple((*bar_a_lines, *bar_b_lines, *bar_a_padding))
```

Comment:
```suggestion
                bar = (*bar_a_lines, *bar_b_lines, *bar_a_padding)
```
It's already a tuple